### ISSUE TITLE
--argfile with invalid input produces an assertion violation 

### ISSUE THREAD

**Issue by pkoppstein** [stedolan/jq#799]
```
$ jq --version
jq-1.5rc1-88-gdff96ae

$ cat array.txt

[
"abc",
"def",
]

$ jq -n --argfile l array.json '$l'
Assertion failed: (jv_is_valid(v)), function jq_next, file execute.c, line 349.
jq: line 2: 12008 Abort trap: 6           ~/github/jq/jq -M "$@"
```
(Note that using jq 1.4 instead, no assertion violation is produced.)

**Comment by nicowilliams** [comment]
Thanks!  Hmm, I made --argfile work as it did in 1.4, but the docs don't describe it well.  I'll have to push a fix for that.

**Comment by nicowilliams** [comment]
Docs fixed.

**Comment by pkoppstein** [comment]
@nicowilliams wrote:
> Docs fixed.

Did you mean to remove --argfile from the -h text?

**Comment by nicowilliams** [comment]
Yes.  I really screwed up when I added it, and since it shipped in 1.4, I can't change its behavior now.  So I added `--slurpfile` instead to replace it.
